docs/deployment/azure/application-insights.md

```diff
@@ -9,60 +9,151 @@ ms.topic: how-to
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddAzureProvisioning();
+
+var insights = builder.AddConnectionString("myInsightsResource", "APPLICATIONINSIGHTS_CONNECTION_STRING");
+
+var apiService = builder.AddProject<Projects.aspire_app11_ApiService>("apiservice")
+    .WithReference(insights);
+
+builder.AddProject<Projects.aspire_app11_Web>("webfrontend")
+    .WithReference(apiService)
+    .WithReference(insights);
+
+builder.Build().Run();
+```
+
+#### Resource usage during developemnt
+
+When running the Aspire application locally, the preceding code reads the connection string from configuration. As this is a secret we should store the value in [app secrets](/aspnet/core/security/app-secrets). Right click on the AppHost project and choose Manage Secrets from the context menu to open the secrets file for the AppHost project. In the file add the key and your specific connection string, the example below is for illustration purposes.
 
 ```json
 {
-  "APPLICATIONINSIGHTS_CONNECTION_STRING": "InstrumentationKey=12345678-abcd-1234-abcd-1234abcd5678;IngestionEndpoint=https://westus3-1.in.applicationinsights.azure.com"
+  "ConnectionStrings": {
+    "myInsightsResource": "InstrumentationKey=12345678-abcd-1234-abcd-1234abcd5678;IngestionEndpoint=https://westus3-1.in.applicationinsights.azure.com"
+  }
 }
 ```
 
+> Note: The keyname specified in the AppHost code needs to match a key inside the `ConnectionStrings` section in the settings file.
+
+#### Resource usage during deployment
+
+When [deploying an Aspire application with Azure Developer CLI (AZD)](./aca-deployment-azd-in-depth.md), it will recognize the connection string resource and prompt for a value. This enables a different resource to be used for the deployment from the value used for local development.
+
+### Mixed deployment
+
+If you wish to use a different mechanism at developemnt from deployment, you can use a combination of Application Insights and Connection String resources, with a conditional. For example the following code uses an automatically provisioned resource at development time, and a pre-supplied connection at deployment time.
+
+```
+var builder = DistributedApplication.CreateBuilder(args);
+
+builder.AddAzureProvisioning();
+
+var insights = builder.ExecutionContext.IsPublishMode ?
+    builder.AddConnectionString("ai", "APPLICATIONINSIGHTS_CONNECTION_STRING") :
+    builder.AddAzureApplicationInsights("ai");
+
+var apiService = builder.AddProject<Projects.aspire_app11_ApiService>("apiservice")
+    .WithReference(insights);
+
+builder.AddProject<Projects.aspire_app11_Web>("webfrontend")
+    .WithReference(apiService)
+    .WithReference(insights);
+
+builder.Build().Run();
+```
+
+> Note: With the above you will need to supply the `Azure:Location` and `Azure:Subscription` information in app secrets for development time usage, and will be prompted for the connection string by AZD at deployment time.
```

Comment:
Please confirm my suggested change here, it's `Azure:SubscriptionId`, right? And not `Azure:Subscription`.

```suggestion
> [!TIP]
> The preceding code requires you to supply the `Azure:Location` and `Azure:SubscriptionId` information in app secrets for development time usage, and will be prompted for the connection string by AZD at deployment time.
```